### DISCUSSION TITLE
New tester or just help? I'm here!

### DISCUSSION THREAD

**Discussion by valerix85** [DurgNomis-drol/mytoyota#310] [General]
Hello everyone and thank you for the immense effort you are making, if you want I can give you a hand with my wife's Yaris Cross and we can debug it to understand the API calls well. Can I do some sniffing on my network to better understand how the calls are made or do you need anything else?

**Comment by CM000n** [comment]
That's good to know @GitOldGrumpy, as soon as we start exploring the get endpoints 😉 
In the meantime, you can simply check the endpoints yourself @valerix85 and report as soon as you discover anything suspicious or an error.

**Comment by valerix85** [comment]
Thanks @CM000n! I will do it with the remote unlock/lock and four arrows (emergency lights). I hope that my iPhone and Nexux 5 can do this job =))

**Comment by CM000n** [comment]
If you already have the current Home Assistant component installed @valerix85, could you post the attributes of the Vin sensor here so we can get an idea of your Yaris' capabilities?

I started also a discussion in the HA Component repo to get an idea which cars with which capapbilites are exist: https://github.com/DurgNomis-drol/ha_toyota/discussions/216
FYI: @GitOldGrumpy & @DurgNomis-drol 😉

**Comment by valerix85** [comment]
Sure! Here some screenshots, in the meantime I've made some test using wireshark and doing the lock feature via App, I will analyze the log this evening.

**Comment by CM000n** [comment]
Hi @valerix85 , thank you very much. I have deleted your screenshots as they also contained the VIN itself. This is something you shouldn't show around as it is the unique ID of your vehicle. 😉 

I also just meant the attributes within the VIN sensor. These provide information about which functions your vehicle has.

Something like this:

```
Contract ID:                        **********23023
IMEI:                               **********07057
Katashiki code:                     ZWE211L-DWXNBW
ASI code:                           FC
Brand:                              Toyota
Car line name:                      Corolla HB/TS
Car model year:                     2019
Car model name:                     Corolla HB/TS - NG '19
Color:                              Emotional Red 2
Generation:                         17CYPLUS
Manufactured date:                  26. Oktober 2020
Date of first use:                  19. November 2020
Transmission type:                  A
Fuel type:                          B
Electrical platform code:           15
EV vehicle:                         false

Features
dealer_appointment:                 true
hybrid_pulse:                       true
last_parked:                        true
privacy:                            true
service_history:                    true
telemetry:                          true
vehicle_diagnostic:                 true
vehicle_status:                     true

Extended capabilities
bonnet_status:                      true
front_driver_door_lock_status:      true
front_driver_door_open_status:      true
front_driver_door_window_status:    true
front_passenger_door_lock_status:   true
front_passenger_door_open_status:   true
front_passenger_door_window_status: true
fuel_level_available:               true
fuel_range_available:               true
hybrid_pulse:                       true
light_status:                       true
rear_driver_door_lock_status:       true
rear_driver_door_open_status:       true
rear_driver_door_window_status:     true
rear_passenger_door_lock_status:    true
rear_passenger_door_open_status:    true
rear_passenger_door_window_status:  true
smart_key_status:                   true
telemetry_capable:                  true
vehicle_diagnostic_capable:         true
vehicle_status:                     true

Remote service capabilities
{}
```

**Comment by valerix85** [comment]
you're right, thanks! I was in a rush this morning to go to work, I didn't notice. Here are the required attributes =)

```
Contract ID
Contract ID:                	      **********68371
IMEI:                       	      **********58597
Katashiki code:             	      MXPJ10L-BHXNBW
ASI code:                   	      FA
Brand:                      	      Toyota
Car line name:              	      Yaris Cross
Car model year:             	      2022
Car model name:             	      Yaris Cross - MY '22
Color:                      	      Grey Metallic
Generation:                 	      21MM
Manufactured date:          	      1 giugno 2023
Date of first use:          	      21 luglio 2023
Transmission type:          	      A
Fuel type:                  	      B
Electrical platform code:   	      19
EV vehicle:                 	      false

Features
dealer_appointment:                   true
hybrid_pulse:                         true
last_parked:                          true
my_destination:                       true
privacy:                              true
remote_service:                       true
service_history:                      true
telemetry:                            true
vehicle_diagnostic:                   true
vehicle_status:                       true
we_hybrid:                            true

Extended capabilities
bonnet_status:                        true
door_lock_unlock_capable:             true
front_driver_door_lock_status:        true
front_driver_door_open_status:        true
front_driver_door_window_status:      true
front_passenger_door_lock_status:     true
front_passenger_door_open_status:     true
front_passenger_door_window_status:   true
fuel_level_available:                 true
fuel_range_available:                 true
guest_driver:                         true
hazard_capable:                       true
hybrid_pulse:                         true
last_parked_capable:                  true
light_status:                         true
rear_driver_door_lock_status:         true
rear_driver_door_open_status:         true
rear_driver_door_window_status:       true
rear_hatch_rear_window:               true
rear_passenger_door_lock_status:      true
rear_passenger_door_open_status:      true
rear_passenger_door_window_status:    true
remote_engine_start_stop:             true
smart_key_status:                     true
telemetry_capable:                    true
vehicle_diagnostic_capable:           true
vehicle_finder:                       true
vehicle_status:                       true
we_hybrid_capable:                    true

Remote service capabilities 
{}
```

**Comment by CM000n** [comment]
Thank you very much 😊 
FYI: I will now convert this issue into a discussion so that we can have a better focus on specific bugs or code adjustments under issues 😉

**Comment by valerix85** [comment]
Thank you too 😄 
I will try to focus on this attribute "remote_service" so we can probably discover the function like "lock/unlock".

**Comment by GitOldGrumpy** [comment]
@valerix85 I suspect the endpoint you are looking for is `/v1/global/remote/command` but have no way to verify this as neither of my cars have the remote lock facility. It will need you to POST requests and the body will contain a JSON object. The JSON object is the bit that will need working out. I'd check back in the old(pre-api change) code as I think it might have been using this endpoint or it could have been the Toyota code for America.

This is how the HVAC stuff works but with a different endpoint I should have that up by the end of the week but I dont know if it will help much.

**Comment by valerix85** [comment]
@GitOldGrumpy I have done yesterday some sniffig using wireshark do you use other tools to see the POST request?

**Comment by GitOldGrumpy** [comment]
Its a pain and hopefully you have an old phone around. Onto which you will need to install PCAPdroid, PCAPdroid mitm. Then you need to convince the MyToyota app to accept your self signed certificate. Then you can use the files create in PCAPdroid in Wireshark.

**Comment by valerix85** [comment]
I've a Nexus 5 to use in this case!

**Comment by valerix85** [comment]
Hi @GitOldGrumpy I tried with my old Nexus 5 but it's too old for the new app, I'm recovering another Android to do so if you can wait more. Thanks!
